### ISSUE TITLE
holding certain keys prevents others from being pressed

### ISSUE THREAD

**Issue by Cooliokid956** [jeremyckahn/keydrown#18]
I come from the same project as the last person (sm64js), and I changed the controls in the game because sometimes some keys don't respond when others are held down, and this can limit your moveset. Do you know why this happens?

**Comment by jeremyckahn** [comment]
Hi @Cooliokid956, thanks for reporting this! Do you know which keys are problematic? And just to be sure, this happens with some keys and not all of them, correct?

**Comment by Cooliokid956** [comment]
Actually, I think it's just because I'm on a Chromebook, sorry to bother, but I'll also try on a Windows computer.

**Comment by jeremyckahn** [comment]
Oh, that's interesting. I don't have a Chromebook, so I can't test that environment. Please let me know if you can reproduce this issue on Windows or any other OS!

**Comment by loglot** [comment]
this is fully a problem with certain keyboards, the Chromebook keyboard isn't sending the keys because it physically can't with how it was made, no amount of software would be able to detect it.

**Comment by jeremyckahn** [comment]
Thanks for the additional info @loglot.

If anyone knows of a way to reproduce this issue in a non-Chromebook environment, I can look into implementing a fix.

**Comment by loglot** [comment]
> If anyone knows of a way to reproduce this issue in a non-Chromebook environment, I can look into implementing a fix.

I have a keyboard that does this with windows, but it happens with all apps, not just keydrown, or even just web browsers, I truly cannot see a fix possible to be implemented

**Comment by Cooliokid956** [comment]
TL;DR: It's the keyboard's fault, not yours

GitHub is informing me of recent activity in this issue, so I'll reinforce the claim made above:

This is a hardware issue stemming from [key rollover](https://en.wikipedia.org/wiki/Key_rollover) where certain keys may be ignored when others are down already.
Some keyboards have "n-key" rollover and will work perfectly (all keystrokes will be registered.)

The quality of the Chromebook's keyboard probably wasn't all that great, which led me to create this issue. Sometime later, I discovered the rollover mechanic and considered following up on this issue but felt too much time had passed.

**Comment by jeremyckahn** [comment]
Thank you for following up and for the clarification, @Cooliokid956!

**Comment by loglot** [comment]
> TL;DR: It's the keyboard's fault, not yours

yeah, that's what I was trying to say lol
